### PR TITLE
feat: Add deployment name and assembled streaming response to chat logs #346

### PR DESCRIPTION
**Pull request by astsiapanay**

### CODE REVIEW

**Comment by astsiapanay** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-347.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by astsiapanay** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-347.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by astsiapanay** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-347.nightly-test.deltixhub.io
>E2E tests status: success

File: src/main/java/com/epam/aidial/core/log/GfLogStore.java [hunk 1]
```diff
@@ -199,4 +216,108 @@ private static String formatTimestamp(long timestamp) {
...
+            message.put("content", content.toString());
+
+            return ProxyUtil.convertToString(result);
+        } catch (Throwable e) {
+            log.warn("Can't assemble streaming response", e);
+            return "{}";
+        }
+    }
+
+    static boolean isStreamingResponse(Buffer response) {
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/log/GfLogStore.java:337]: please, add a comment with examples, hard to understand what is streaming and non-streaming format you are checking

File: src/main/java/com/epam/aidial/core/log/GfLogStore.java [hunk 2]
```diff
@@ -199,4 +216,108 @@ private static String formatTimestamp(long timestamp) {
         return LocalDateTime.ofInstant(Instant.ofEpochMilli(timestamp), ZoneId.of("UTC"))
                 .format(DateTimeFormatter.ISO_DATE_TIME);
     }
+
+    static String assembleStreamingResponse(Buffer response) {
+        try (Scanner scanner = new Scanner(new ByteBufInputStream(response.getByteBuf()))) {
+            StringBuilder content = new StringBuilder();
+            ObjectNode result = ProxyUtil.MAPPER.createObjectNode();
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/log/GfLogStore.java]: Nit. can me moved to the line 280

File: src/main/java/com/epam/aidial/core/log/GfLogStore.java [hunk 3]
```diff
@@ -199,4 +216,108 @@ private static String formatTimestamp(long timestamp) {
...
+            ObjectNode last = null;
+            ObjectNode choice = ProxyUtil.MAPPER.createObjectNode();
+            ObjectNode message = ProxyUtil.MAPPER.createObjectNode();
+            choice.set("message", message);
+            JsonNode usage = null;
+            JsonNode statistics = null;
+            JsonNode systemFingerprint = null;
+            JsonNode model = null;
+            scanner.useDelimiter("\n*data: *");
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/log/GfLogStore.java:239]: it worth to add a comment for delimiter pattern as well

File: src/main/java/com/epam/aidial/core/log/GfLogStore.java (hunk 1 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/log/GfLogStore.java:337]: fixed.

File: src/main/java/com/epam/aidial/core/log/GfLogStore.java (hunk 2 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/log/GfLogStore.java]: fixed

File: src/main/java/com/epam/aidial/core/log/GfLogStore.java (hunk 3 above)
Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/log/GfLogStore.java:239]: fixed

**Review by Maxim-Gadalov — APPROVED** [review]

**Pull request merged**